mods/ra/maps/monster-tank-madness/monster-tank-madness.lua

```diff
@@ -103,13 +103,35 @@ SendAlliedUnits = function()
 	Media.PlaySpeechNotification(player, "ReinforcementsArrived")
 	Utils.Do(AlliedUnits, function(table)
 		Trigger.AfterDelay(table.delay, function()
-			Reinforcements.Reinforce(player, table.types, { StartEntryPoint.Location, StartMovePoint.Location }, 18)
+			local units = Reinforcements.Reinforce(player, table.types, { StartEntryPoint.Location, StartMovePoint.Location }, 18)
+
+			Utils.Do(units, function(unit)
+				if unit.Type == "e6" then
+					Trigger.OnKilled(unit, LandingPossible)
+				end
+			end)
 		end)
 	end)
 
 	Trigger.AfterDelay(DateTime.Seconds(1), function() InitialUnitsArrived = true end)
 end
 
+LandingPossible = function()
+	if not beachReached then
+		fail = true
+		Utils.Do(Map.ActorsInBox(Map.TopLeft, Map.BottomRight), function(a)
+			if a.Owner == player then
+				if a.Type == "spen" or a.Type == "lst" then
+					fail = false
```

Comment:
Can't you just avoid the `fail` variable (which is apparently global, btw)  and replace this line with `player.MarkFailedObjective(CrossRiver)` (and possibly immediately `return`)?